validation_framework/deployer/coe/docker.py

```diff
@@ -0,0 +1,239 @@
+                         f'\n\n {json.dumps(envs_configuration, indent=4)} \n')
+        self.device.run_command(start_command, envs=envs_configuration)
+        self.logger.info('Device start command executed')
+
+    def add_peripheral(self):
+        pass
+
+    def remove_peripheral(self):
+        pass
+
+    def stop_engine(self):
+        self.purge_engine()
+
+    def get_authorized_keys(self):
+        return self.device.run_command("cat ~/.ssh/authorized_keys")
+
+    def peripherals_running(self, peripherals: set) -> bool:
+        cont_filter: dict = {"Names": "{{ .Names }}"}
+
+        data: list[dict] = self.device.get_remote_containers(
+            containers_filter=cont_filter)
+
+        # Find network container in list
+        self.logger.debug('Removing ')
+        for c in data:
+            if peripherals.__contains__(c):
+                peripherals.remove(c.get("Names"))
+        return not bool(peripherals)
+
+    def get_remote_containers(self, containers_filter: dict = None) -> list:
+        return self.device.get_remote_containers(containers_filter)
+
+    def get_engine_logs(self):
+        self.logger.info(f'Retrieving Log files from engine run with UUID: {self.engine_configuration.nuvlaedge_uuid}')
+        # 1. Retrieve deployment containers ID's
+        result: Result = self.device.run_command(
+            "docker ps -a --format '{\"ID\":\"{{ .ID }}\", \"Image\": \"{{ .Image }}\", \"Names\":\"{{ .Names }}\"}' "
+            f"--filter label=com.docker.compose.project={self.engine_configuration.compose_project_name} | jq --tab -s .")
+
+        self.logger.debug(f'Extracting logs from: \n\n\t{result.stdout}\n')
+        containers = json.loads(result.stdout)
+
+        # 2. Iteratively copy files from /var/docker/logs/<container_id>.log to
+        # /tmp/<new_folder> and chmod before transferring it the running machine
+        self.device.run_command(
+            f'mkdir -p /tmp/{self.engine_configuration.compose_project_name}')
+        local_tmp_path: Path = Path(f'/tmp/{self.engine_configuration.compose_project_name}')
+        local_tmp_path.mkdir(parents=True, exist_ok=True)
+
+        for c in containers:
+            c_name = c.get('Names')
+            full_id = self.device.run_command(command='docker inspect --format="{{.Id}}" ' + c_name).stdout
+            full_id = full_id.replace('\n', '')
+            full_id = full_id.replace('\\', '')
+
+            self.logger.debug(f'Processing logs for nuvlaedge {c_name}')
+            self.device.run_sudo_command(f'sudo cp /var/lib/docker/containers/{full_id}/{full_id}-json.log '
+                                         f'/tmp/{c_name}.log')
+            self.device.run_sudo_command(f'sudo chmod 777 /tmp/{c_name}.log')
+
+            # 3. Transfer back the files
+
+            self.device.download_remote_file(remote_file_path=f'/tmp/{c_name}.log',
+                                             local_file_path=local_tmp_path / (c_name + '.log'))
```

Comment:
Why not splitting this into another method? It's usually better to keep the methods/functions as simple as possible to facilitate testing. The get_container_logs comes in handy

Reply:
Done.